utility_network/display-utility-associations/src/main/java/com/esri/samples/display_utility_associations/DisplayUtilityAssociationsSample.java

```diff
@@ -0,0 +1,245 @@
+        }
+      });
+  
+      // get the symbol from the connectivitySymbol
+      ListenableFuture<Image> connectivityBitmap = connectivitySymbol.createSwatchAsync(0x00000000);
+      connectivityBitmap.addDoneListener(() -> {
+        try {
+          // display the image view in the preview area
+          connectivityImageView.setImage(connectivityBitmap.get());
+        } catch (InterruptedException | ExecutionException e) {
+          new Alert(Alert.AlertType.ERROR, "Error creating preview ImageView from provided connectivitySymbol" + e.getMessage()).show();
+        }
+      });
+      
+      
+      // Create a renderer for the associations.
+      UniqueValueRenderer.UniqueValue attachmentUV = new UniqueValueRenderer.UniqueValue("Attachment", "", attachmentSymbol, Arrays.asList(UtilityAssociationType.ATTACHMENT.toString()));
+      UniqueValueRenderer.UniqueValue connectivityUV = new UniqueValueRenderer.UniqueValue("Connectivity", "", connectivitySymbol, Arrays.asList(UtilityAssociationType.CONNECTIVITY.toString()));
+      associationsOverlay.setRenderer(new UniqueValueRenderer(Arrays.asList("AssociationType"), Arrays.asList(attachmentUV, connectivityUV), "", null));
+      
+      // set initial ArcGISMap extent
+      map.setInitialViewpoint(viewPoint);
+      // create a map view and set its map
+      mapView = new MapView();
+      mapView.setMap(map);
+      mapView.getGraphicsOverlays().add(associationsOverlay);
+      
+      // create a controls box
+      VBox controlsVBox = new VBox();
+      controlsVBox.setBackground(new Background(new BackgroundFill(Paint.valueOf("rgba(255,255,255,0.5)"), CornerRadii.EMPTY,
+        Insets.EMPTY)));
+      controlsVBox.setPadding(new Insets(10.0));
+      controlsVBox.setSpacing(5);
+      controlsVBox.setMaxSize(100, 50);
+      controlsVBox.setAlignment(Pos.BASELINE_LEFT);
+      controlsVBox.getStyleClass().add("panel-region");
+      controlsVBox.getChildren().addAll(attachmentLabel, attachmentImageView, connectivityLabel, connectivityImageView);
+      
+      // add the scene view and controls to the stack pane
+      stackPane.getChildren().addAll(mapView, controlsVBox);
+      StackPane.setAlignment(controlsVBox, Pos.TOP_LEFT);
+      StackPane.setMargin(controlsVBox, new Insets(10, 0, 0, 10));
+      
+      
+    } catch (Exception e) {
+      // on any error, display the stack trace.
+      e.printStackTrace();
+    }
+    
+  }
+  
+  private void addAssociation() {
+    try {
+      // Check if the current viewpoint is outside of the max scale.
+      if (mapView.getCurrentViewpoint(Viewpoint.Type.CENTER_AND_SCALE).getTargetScale() >= maxScale)
+        return;
+      
+      // Check if the current viewpoint has an extent.
+      Envelope extent = mapView.getCurrentViewpoint(Viewpoint.Type.BOUNDING_GEOMETRY).getTargetGeometry().getExtent();
+      if (extent == null) {
+        return;
+      }
```

Comment:
This is an example where we could do it the 'java-way':
```suggestion
      if (extent != null) {
        // do the rest of the logic (lines 202- 232)
      }
```